### PR TITLE
Fix all tests + first pass at CI

### PR DESCRIPTION
**Pull request by ardila**
* Switched to public URLs (hosted in this github) in order to avoid the need for okta credentials
* Added timestamp to model reference to fix some broken tests, now all tests pass again
* Checked in first attempt at circleci config, borrowing from the scaleapi python client.
* Fixed black and pylint issues
* Fixed what appeared to be broken tests for ignore (that expected ignored stuff to be processed?)
* Re-add support for python 3.6 via dataclasses package
* Switch to testing against python 3.6

Primary reviewer: Srikanth

### CODE REVIEW

File: nucleus/annotation.py [hunk 1]
```diff
@@ -21,10 +22,11 @@
     ANNOTATIONS_KEY,
 )
 
-from dataclasses import dataclass
-
 
 class Annotation:
+    reference_id: Optional[str] = None
```
Comment (ardila) [inline nucleus/annotation.py:27]: Added this to avoid pylint

File: nucleus/dataset_item.py [hunk 2]
```diff
@@ -7,16 +9,14 @@
...
-
 
 @dataclass
 class DatasetItem:
 
     image_location: str
-    reference_id: str = None
-    item_id: str = None
-    metadata: dict = None
+    reference_id: Optional[str] = None
```
Comment (ardila) [inline nucleus/dataset_item.py:17]: fix typing mistake

File: nucleus/prediction.py [hunk 3]
```diff
@@ -58,15 +58,15 @@ def __init__(
...
-            label,
-            x,
-            y,
-            width,
-            height,
-            reference_id,
-            item_id,
-            annotation_id,
-            metadata,
+            label=label,
```
Comment (ardila) [inline nucleus/prediction.py:61]: added keywords to avoid ordering bugs

File: nucleus/prediction.py [hunk 4]
```diff
@@ -106,7 +106,12 @@ def __init__(
         metadata: Optional[Dict] = None,
     ):
         super().__init__(
-            label, vertices, reference_id, item_id, annotation_id, metadata
+            label=label,
```
Comment (ardila) [inline nucleus/prediction.py:109]: added keywords to avoid ordering bugs

File: nucleus/slice.py [hunk 5]
```diff
@@ -9,7 +9,7 @@ class Slice:
     def __init__(self, slice_id: str, client):
         self.slice_id = slice_id
         self._client = client
-    
+
```
Comment (ardila) [inline nucleus/slice.py:12]: black

File: tests/test_dataset.py [hunk 6]
```diff
@@ -19,26 +19,27 @@
...
-                    "example_dict": {
-                        "nested": True,
-                    },
-                    "example_list": ["hello", 1, False],
+
+    test_repr(
+        DatasetItem(
+            image_location="test_url",
+            reference_id="test_reference_id",
+            metadata={
```
Comment (ardila) [inline tests/test_dataset.py:31]: black

File: tests/test_models.py [hunk 7]
```diff
@@ -19,28 +20,35 @@
...
+
     client = NucleusClient(api_key="fake_key")
-    test_repr(Model(client=client, model_id="fake_model_id", name="fake_name", reference_id="fake_reference_id", metadata={"fake": "metadata"}))
+    test_repr(
+        Model(
+            client=client,
+            model_id="fake_model_id",
+            name="fake_name",
+            reference_id="fake_reference_id",
+            metadata={"fake": "metadata"},
```
Comment (ardila) [inline tests/test_models.py:40]: black

**Review by srikanth-scale-2 — APPROVED** [review]
This is sick, excited for CI 🎉

File: tests/helpers.py [hunk 8]
```diff
@@ -1,28 +1,25 @@
...
-TEST_DATASET_NAME = "[PyTest] Test Dataset"
-TEST_SLICE_NAME = "[PyTest] Test Slice"
+
 TEST_IMG_URLS = [
-    "s3://scaleapi-cust-lidar/Hesai/raw_data/2019-5-11/hesai_data_1557540003/undistorted/front_camera/1557540143.650423lf.jpg",
-    "s3://scaleapi-cust-lidar/Hesai/raw_data/2019-5-11/hesai_data_1557540003/undistorted/back_camera/1557540143.600352lf.jpg",
-    "s3://scaleapi-cust-lidar/Hesai/raw_data/2019-5-11/hesai_data_1557540003/undistorted/right_camera/1557540143.681730lf.jpg",
-    "s3://scaleapi-cust-lidar/Hesai/raw_data/2019-5-11/hesai_data_1557540003/undistorted/front_left_camera/1557540143.639619lf.jpg",
-    "s3://scaleapi-cust-lidar/Hesai/raw_data/2019-5-11/hesai_data_1557540003/undistorted/front_right_camera/1557540143.661212lf.jpg",
+    "http://farm1.staticflickr.com/107/309278012_7a1f67deaa_z.jpg",
```
Comment (srikanth-scale-2) [inline tests/helpers.py:16]: Why are we not using s3?

Comment (sasha-scale) [reply tests/helpers.py:16]: to avoid dependency on AWS creds in CircleCI

**Pull request merged**